ops/pebble.py

```diff
@@ -2335,6 +2369,19 @@ def _wait_change_using_polling(
 
         raise TimeoutError(f'timed out waiting for change {change_id} ({timeout} seconds)')
 
+    def _checks_action(self, action: str, checks: Iterable[str]) -> List[str]:
+        if isinstance(checks, (str, bytes)) or not hasattr(checks, '__iter__'):
+            raise TypeError(f'checks must be of type Iterable[str], not {type(checks).__name__}')
+
+        checks = tuple(checks)
+        for chk in checks:
+            if not isinstance(chk, str):
+                raise TypeError(f'check names must be str, not {type(chk).__name__}')
+
+        body = {'action': action, 'checks': checks}
+        resp = self._request('POST', '/v1/checks', body=body)
```

Comment:
P.S. If any check name is actually bytes, it will not get serialised as JSON, raising a TypeError instead.

Reply:
That couldn't happen, though, because there's a check for each individual check name being a string above this (so that you get an error if you pass in `list[int]` or something).